cmd/oras/internal/display/metadata/view/printer.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package view
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io"
+
+	"github.com/Masterminds/sprig/v3"
+)
+
+// Outputable interface is used to reset output of the handler.
+type Outputable interface {
+	// WithOutput resets output of the handler.
+	WithOutput(out io.Writer)
+}
+
+// Printer prints.
+type Printer interface {
+	Printf(format string, a ...any) (n int, err error)
+	Println(a ...any) (n int, err error)
+	PrintJSON(object any) error
```

Comment:
I was thinking the printer wouldn't know about format, but I'll have to think about that more. 

I was in the process of unwinding `verbose` everywhere which is a lot more complicated than this.

Reply:
`verbose` serves only for **status output**. E.g. in the below output:
```console
➜  output oras push localhost:5000/demo:output image.tar --no-tty
Uploading 8b138a688d47 image.tar
Uploaded  8b138a688d47 image.tar
Pushed [registry] localhost:5000/demo:output
Digest: sha256:b7593bbc81769aa9fef0b37b73405eccfaf968a56e6b542614cfb19fe376e782
```
The status output is below and only covers **named** content
```console
Uploading 8b138a688d47 image.tar
Uploaded  8b138a688d47 image.tar
```

If we run it in verbose mode, then the output is
```console
➜  output oras push localhost:5000/demo:output image.tar --no-tty --verbose
Preparing image.tar
Uploading 8b138a688d47 image.tar
Uploading 44136fa355b3 application/vnd.oci.empty.v1+json
Uploaded  8b138a688d47 image.tar
Uploaded  44136fa355b3 application/vnd.oci.empty.v1+json
Uploading 262c1a7d7850 application/vnd.oci.image.manifest.v1+json
Uploaded  262c1a7d7850 application/vnd.oci.image.manifest.v1+json
Pushed [registry] localhost:5000/demo:output
Digest: sha256:262c1a7d7850ee622859898dafa7a13618fabb9beb5b4fa5dfb8cc1e73bbb8d7
```
In this case, also unnamed content is output:
```console
Uploading 8b138a688d47 image.tar
Uploading 44136fa355b3 application/vnd.oci.empty.v1+json
Uploaded  8b138a688d47 image.tar
Uploaded  44136fa355b3 application/vnd.oci.empty.v1+json
Uploading 262c1a7d7850 application/vnd.oci.image.manifest.v1+json
Uploaded  262c1a7d7850 application/vnd.oci.image.manifest.v1+json
```

Reply:
This PR only focus on **metadata output**, in the above case, the metadata output is:
```console
Pushed [registry] localhost:5000/demo:output
Digest: sha256:262c1a7d7850ee622859898dafa7a13618fabb9beb5b4fa5dfb8cc1e73bbb8d7
```

and 

```console
Pushed [registry] localhost:5000/demo:output
Digest: sha256:b7593bbc81769aa9fef0b37b73405eccfaf968a56e6b542614cfb19fe376e782
```